apps/server/src/modules/board/uc/element.uc.ts

```diff
@@ -55,7 +55,14 @@ export class ElementUc {
 			);
 		}
 
-		await this.checkPermission(userId, submissionContainer, Action.write);
+		const boardDoAuthorizable = await this.boardDoAuthorizableService.getBoardAuthorizable(submissionContainer);
+		const boardRoles = boardDoAuthorizable.users.find((user) => user.userId === userId)?.roles;
```

Comment:
It look like a rule for the authorization service.

Reply:
This is not clear to me yet.

The BoradDORule has a single hasPermisison method which checks if the user is reader or editor.
In this case we want to prevent teacher aka. editor and only allow readers (students) to pass.

Not sure where and how to implement that without added too much complexity for an edge case.

Here I solved it with 3 lines. I would very much prefer this than to introduce a new rule or even extend the rule engine.

Reply:
I updated the BoardRule to have an optional UserRoleEnum and if present the user must conform to that.
That means we can limit the authorization to either Student, Teacher or SubstituteTeacher.